cuqi/solver/_solver.py

```diff
@@ -650,8 +650,124 @@ def solve(self):
                 x_new = x_new + ((k-1)/(k+2))*(x_new - x_old)
               
             x = x_new.copy()
+
+class ADMM(object):
+    """Alternating Direction Method of Multipliers for solving regularized linear least squares problems of the form:
+    Minimize ||Ax-b||^2 + sum_i f_i(L_i x).
+
+    Reference:
+    Boyd et al. "Distributed optimization and statistical learning via the alternating direction method of multipliers."Foundations and Trends® in Machine learning, 2011.
+
     
+    Parameters
+    ----------
+    A : ndarray or callable f(x,*args).
```

Comment:
I think "f" is a bit misleading here. Maybe just do this
```suggestion
    A : ndarray or callable with arguments (x, *args).
```

Reply:
I think I copied this line directly form the CGLS or FISTA documentation, so such update would have to be done to all.

I agree that removing the explicit mention of f could help. Perhaps we could also explicitly mention what the arguments are, i.e., the flag parameter and what is expected of this. Do you have any suggestions of how this can be done properly in the documentation?

Reply:
Something like
```suggestion
    A : ndarray or callable
        Represents a matrix or a function that performs matrix-vector multiplications. 
        When A is a callable, it accepts arguments (x, flag) where:
        - flag=1 indicates multiplication of A with vector x, that is A @ x.
        - flag=2 indicates multiplication of the transpose of A with vector x, that is  A.T @ x.
```